hydrofunctions/hydrofunctions.py

```diff
@@ -122,14 +138,58 @@ def extract_nwis_df(response_obj):
         raise exceptions.HydroNoDataError("The NWIS reports that it does not"
                                           " have any data for this request.")
 
-    data = nwis_dict['value']['timeSeries'][0]['values'][0]['value']
-
+    # create lists of timeseries keys, names, and noDataValues
```

Comment:
This is so complex! It is a vast improvement over what I had. 

- `keys` is a list where each element matches a requested time series.
- `tts` is a time series.
- `names` is a list of site names that corresponds with `keys`
- `noDataValues` is a list of all of the noDataValues that can be found in our response. I've been assuming that it always is '-999999', but that is a bad assumption...

Reply:
So far I have only encountered `noDataValues = -999999` but according to one of our NWIS people `noDataValues` it is not always equal to `-999999`.

* Might also be nice to add helper functions to get full site name, statistic, units, etc. from json. If we did this we could used the defined `name` for a data column which is a combination of the data provider (`USGS`), site id, parameterCd, and statistic (for example, `USGS:123456:00060:001`). 
* At some point could add the bounding box option.

I could implement the helper functions and modified name prior to the release you are working towards. Let me know.

Reply:
These both sound like great ideas. You could create a new branch from Develop that adds the helper functions and the bounding box. The helper functions could store the metadata in properties that belong to the NWIS object instead of storing them in the dataframe, since extra stuff that gets added onto dataframes often gets lost. Apparently some functions will modify a dataframe, and other will just create a new dataframe.

The bounding box idea could be added as a keyword argument to the .get_nwis() method of hydrofunctions. I was considering adding a 'period' keyword argument to .get_nwis() as well. 

Finally, I was thinking that it was stupid that you have to run hf.NWIS.get_data() to retrieve data from NWIS, and run .df() every time you want a dataframe. Maybe instead I would set it up so that the data are retrieved during NWIS.init() when the object is created.

Anyway, maybe all of this new functionality could be added **after** we push out a new version to PyPI. That way we don't need to worry about completing everything before you have your class!!

Thanks!
-Marty